site/content/contribute/webapp/migrating-to-typescript.md

```diff
@@ -0,0 +1,38 @@
+---
+title: "Migrating to Typescript"
+date: 2020-07-11T23:00:00-04:00
+weight: 9
+subsection: Web App
+---
+
+The Mattermost team wants to improve the quality, security and stability of the code, and one way to do this is by introducing the usage of type checking. Thus, we have decided to introduce Typescript in our codebase. As a first step, we have migrated the [mattermost-redux](https://github.com/mattermost/mattermost-redux) library to use Typescript, and are now in the processs of migrating the [webapp](https://github.com/mattermost/mattermost-webapp) to use Typescript.
```

Comment:
```suggestion
The Mattermost team wants to improve the quality, security, and stability of the code, and one way to do this is by introducing the usage of type checking. Thus, we have decided to introduce Typescript in our codebase. As a first step, we have migrated the [mattermost-redux](https://github.com/mattermost/mattermost-redux) library to use Typescript, and are now in the process of migrating the [webapp](https://github.com/mattermost/mattermost-webapp) library to use Typescript.
```

Reply:
Three questions:
1. What is "the code"? Is it the entire codebase or a sub-set? 
2. Are there any benefits that can be provided as to why Typescript is the choice? 
3. Were there specific security issues or is this a proactive approach?

Reply:
1. The relevant code is scoped to the webapp repo https://github.com/mattermost/mattermost-webapp.
2. I would say it is the most standard approach today, for adding types to Javascript. It is very mature and feature-rich.
3. It is purely a proactive approach.

I'm 0/5 on not adding the word `library` to the last sentence. The webapp (Web Application) is an application, whereas libraries usually refer to pieces of software that can be used by applications. I think it is still technically a library in nature, but it seems out of place here.

Reply:
Thanks, @mickmister. I've made some changes. LMK what you think? 

Thanks for clarifying re: ``library``. Makes perfect sense and I've removed it. :)